src/nb_ndarray.cpp

```diff
@@ -211,6 +211,9 @@ static PyObject *dlpack_from_buffer_protocol(PyObject *o, bool ro) {
             case 'f':
             case 'd': dt.code = (uint8_t) dlpack::dtype_code::Float; break;
 
+            // case 'Zf':
```

Comment:
This needs looking at since for complex the id isn't a single `char`.

Reply:
Before line 191, add:

```cpp
bool is_complex = format_str[0] == 'Z';
if (is_complex)
    format_str++;
```
Then remove the two 'Z' cases and below the switch statement, add:

```cpp
if (is_complex) {
    fail =  dt.code != (uint8_t) dlpack::dtype_code::Float;
    dt.code = dlpack::dtype_code::Complex;
}
```